Tophat/Extensions/String+Extensions.swift

```diff
@@ -0,0 +1,40 @@
+//
+//  String+Extensions.swift
+//  Tophat
+//
+//  Created by Lukas Romsicki on 2024-12-02.
+//  Copyright © 2024 Shopify. All rights reserved.
+//
+
+import Foundation
+
+extension String {
+	var indefiniteArticle: String {
+		startsWithVowel ? "an" : "a"
+	}
+}
+
+extension String {
+	var isValidURL: Bool {
+		if let detector = try? NSDataDetector(types: NSTextCheckingResult.CheckingType.link.rawValue),
+		   let match = detector.firstMatch(in: self, options: [], range: NSRange(location: 0, length: self.utf16.count)) {
+			return match.range.length == self.utf16.count
+		}
+
+		return false
+	}
+}
+
+extension Character {
+	var isVowel: Bool {
+		"aeiou".contains {
```

Comment:
`.caseInsensitive` below means we don't have to include `AEIOU` here, right?

Reply:
Yeah, exactly.